Refactor: Extract Sleep timer

pichfl opened the pull request:
First part of the refactoring effort that builds up to the new layout.

Re #44

pichfl commented:
Do not merge yet, this has a few kinks left, ie. when you turn off the time the label does not reset (which it did in a previous iteration, but I tried to be too clever and sourced the original label text from the current label, which is obviously wrong)

pichfl commented:
Found and fixed my problem, regarding the label, feel free to throw more comments at me when doing a review 🙂

GianniCarlo commented on Audiobook Player/Player/SleepTimer.swift:
```diff
@@ -0,0 +1,138 @@
+
+    let durationFormatter = DateComponentsFormatter()
+
+    var timer: Timer!
```
what do you think about having this var as either:
`lazy var timer = Timer()`
or
`var timer: Timer?`
it would remove the `nil` checks and make the code look more 'swifty'(?). Let me know what you think

pichfl replied:
Still learning swift, sounds good to me.

GianniCarlo commented:
Tried out the app on this branch, I found two things:
- When turning off the timer, playback stops
- If I set the timer, dismiss the player, open it again, the timer label doesn't update itself
Regarding this last point, wouldn't it be weird to have two timers in the screen at once? (playback time and sleep timer)

GianniCarlo commented:
there might be an `IBAction` not connected well for `setSleepTimer` in `PlayerViewController` 🤔 I had to connect it to get the timer button to work, otherwise it was crashing

pichfl commented:
Pretty sure the disconnect happened between one of the refactoring steps. Will fix. 

Playback stopping when disabling the timer was unintentional, will investigate. 

Regarding the label: yes, makes sense, I forgot to set the initial state correctly. 

Regarding the two timers: we can put it back into the message, I just liked that you can see what’s going on right away. I don’t fear that people will mistake either of the fields as they have distinct positions and colors in the redesign.

pichfl commented:
I should have addressed all the issues with the first iteration of this PR. The time left is back in the message of the action sheet to keep this iteration closer to what is in the current version. I left the option to change the label in as well but commented it out.

GianniCarlo commented on Audiobook Player/Player/PlayerViewController.swift:
```diff
@@ -220,128 +209,44 @@ class PlayerViewController: UIViewController {
-            self.sleep(in: 600)
-        }))
+    @IBAction func setSleepTimer() {
+        let actions = SleepTimer.shared.actionSheet(
```
This should be called something along the lines of `actionSheet` since `actions` is not an array of actions, but an instance of `UIAlertController`

GianniCarlo commented on Audiobook Player/Player/SleepTimer.swift:
```diff
@@ -0,0 +1,118 @@
+    private func reset() {
+        alert.message = defaultMessage
+        
+        guard timer != nil else {
```
this shouldn't be necessary, that line after this: `timer?.invalidate()` should only work if timer is != nil, otherwise it won't do anything

GianniCarlo commented:
👍 Awesome!
